File: src/main/java/omerozturk/hrms/business/abstracts/JobPositionService.java
package omerozturk.hrms.business.abstracts;

import omerozturk.hrms.core.utilities.result.DataResult;
import omerozturk.hrms.core.utilities.result.Result;
import omerozturk.hrms.entities.concretes.JobPosition;

import java.util.List;
import java.util.Optional;

public interface JobPositionService {
    DataResult<List<JobPosition>> getAll();
    Result add(JobPosition jobPosition);
    Result delete(JobPosition jobPosition);
    DataResult<JobPosition> getById(int jobPositionId);
}

File: src/main/java/omerozturk/hrms/uiPresentation/src/services/systemServices/jobPositionService.js
import axios from "axios";

export default class JobPositionService {
    getAll() {
        return axios.get("http://localhost:8080/api/jobpositions/getall" );
    }
    getById(id) {
        return axios.get("http://localhost:8080/api/jobpositions/getbyid?jobPositionId=" + id);
    }

    add(jobPosition) {
        return axios.post('http://localhost:8080/api/jobpositions/add', jobPosition)
            .then(function (response) { console.log(response); })
            .catch(function (error) {
                console.log(error);
            });
    }
}